code/src/nuvla/job/actions/nuvlabox_log_fetch.py

```diff
@@ -20,12 +21,29 @@ def all_components(self):
 
     @property
     def connector(self):
+
         if not self._connector:
-            self._connector = \
-                nb.NuvlaBox(
-                    api=self.api,
-                    nuvlabox_id=self.resource_log_parent,
-                    job=self.job)
+            if os.getenv('KUBERNETES_SERVICE_HOST'):
+                path = '/srv/nuvlaedge/shared' # TODO parameterize this
+                ca=open(f'{path}/ca.pem', encoding="utf8").read()
+                key=open(f'{path}/key.pem', encoding="utf8").read()
+                cert=open(f'{path}/cert.pem', encoding="utf8").read()
+                kubernetes_host = os.getenv('KUBERNETES_SERVICE_HOST')
+                kubernetes_port = os.getenv('KUBERNETES_SERVICE_PORT')
+                if kubernetes_host and kubernetes_port:
+                    endpoint=f'https://{kubernetes_host}:{kubernetes_port}'
```

Comment:
Could this be extracted somewhere ? Like in kubernetes.py.
Is it possible to get this information from Nuvla.io ?
If not, a check should be added to ensure we are in `pull` mode because these files are not available on the Nuvla own job executor.

Reply:
I have created a new K8sLogging class to solve all this mess...